store/store.go

```diff
@@ -174,6 +177,27 @@ type Store struct {
 	suggestedCurrency string
 }
 
+func shouldRetryHttpResponse(resp *http.Response, attempt *retry.Attempt) bool {
+	return (resp.StatusCode == 500 || resp.StatusCode == 503) && attempt.More()
+}
+
+func shouldRetryError(err error, attempt *retry.Attempt) bool {
+	if !attempt.More() {
+		return false
+	}
+	if netErr, ok := err.(net.Error); ok {
+		return netErr.Timeout()
+	}
+	return err == io.ErrUnexpectedEOF
+}
+
+var defaultRetryStrategy = retry.LimitCount(6, retry.LimitTime(30*time.Second,
```

Comment:
Perhaps we should start a bit more conservative on the timing. In my vague experience around 10 seconds without proper feedback people start to think the procedure is broken rather than just taking a while. Can we start there?